cpp/src/defaults.cpp

```
@@ -91,6 +91,25 @@ CompatMode getenv_or(std::string_view env_var_name, CompatMode default_val)
   return detail::parse_compat_mode_str(env_val);
 }
 
+template <>
+std::vector<int> getenv_or(std::string_view env_var_name, std::vector<int> default_val)
+{
+  auto* env_val = std::getenv(env_var_name.data());
+  if (env_val == nullptr) { return default_val; }
+  std::vector<int> vect;
+  std::stringstream ss(env_val);
+
+  for (int i; ss >> i;) {
```

My lack of C++ experience is showing here, but I could use some help here. To goal here is to parse a string like `"429,500,503"` to a list of integers like `[429, 500, 503]`.

Right now, it seems like exceptions are being swallowed somwhere, at least when accessed through Python:

```
KVIKIO_HTTP_STATUS_CODES="a" python -c 'import kvikio.defaults; print(kvikio.defaults.http_status_codes())'
[]
```

And we aren't handling valid lists properly :/

```
KVIKIO_HTTP_STATUS_CODES="429,500" python -c 'import kvikio.defaults; print(kvikio.defaults.http_status_codes())'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/coder/kvikio/python/kvikio/kvikio/defaults.py", line 318, in http_status_codes
    return kvikio._lib.defaults.http_status_codes()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "defaults.pyx", line 89, in kvikio._lib.defaults.http_status_codes
ValueError: Invalid value 429,500
```